File: trie/c/test_trie.c
#include <assert.h>

#include "trie.h"

void test_insert_search()
{
	char* setA[24] = { "a", "a", "aa", "aal", "aalii", "aam", "aani",
		"aardvark", "aardwolf", "aaron", "aaronic", "aaronical", "aaronite",
		"aaronitic", "aaru", "ab", "aba", "ababdeh", "ababua", "abac", "abaca",
		"abacate", "abacay", "abacinate"
	};
	char* setB[24] = { "b", "b", "ba", "baa", "baahling", "baal", "baal",
		"baalath", "baalish", "baalism", "baalist", "baalite", "baalitical",
		"baalize", "baalshem", "baar", "bab", "baba", "babacoote", "babai",
		"babasco", "babassu", "babaylan", "babbie"
	};
	Trie trie;
	trie_init(&trie);
	int i;
	for(i = 0; i < 24; i++)
		trie.insert(&trie, setA[i]);
	for(i = 0; i < 24; i++)
		assert(trie.search(&trie, setA[i]));
	for(i = 0; i < 24; i++)
		assert(!trie.search(&trie, setB[i]));
	trie.destroy(&trie);
}

void test_start_with()
{
	char* words[13] = { "aalii",  "aam",  "aani", "aardvark", "aardwolf",
		"aaronic", "aaronite", "aaronitic", "aaru", "ababdeh", "ababua",
		"abacay", "abacinate"
	};
	char* prefixes[8] = {"a","a","aa","aal","aaron","ab","aba","abac"};
	char* others[6] = {
		"abaciscus","abacist","aback","abactinal","Abe","abaction"
	};
	Trie trie;
	trie_init(&trie);
	int i;
	for(i = 0; i < 13; i++)
		trie.insert(&trie, words[i]);
	for(i = 0; i < 8; i++)
		assert(trie.startWith(&trie, prefixes[i]));
	for(i = 0; i < 6; i++)
		assert(!trie.startWith(&trie, others[i]));
	trie.destroy(&trie);
}

int main()
{
	test_insert_search();
	test_start_with();
	return 0;
}

File: trie/java/TestTrie.java

public class TestTrie
{
	public static void testInsertSearch()
	{
		String setA[] = { "A", "a", "aa", "aal", "aalii", "aam", "Aani",
			"aardvark", "aardwolf", "Aaron", "Aaronic", "Aaronical", "Aaronite",
			"Aaronitic", "Aaru", "Ab", "aba", "Ababdeh", "Ababua", "abac", "abaca",
			"abacate", "abacay", "abacinate"
		};
		String setB[] = { "B", "b", "ba", "baa", "baahling", "Baal", "baal",
			"Baalath", "Baalish", "Baalism", "Baalist", "Baalite", "Baalitical",
			"Baalize", "Baalshem", "baar", "Bab", "baba", "babacoote", "babai",
			"babasco", "babassu", "babaylan", "Babbie"
		};
		Trie trie = new Trie();
		for(int i = 0; i < setA.length; i++)
			trie.insert(setA[i]);
		for(int i = 0; i < setA.length; i++)
			if(trie.search(setA[i]) != true)
				System.out.println("Error in insert search");
		for(int i = 0; i < setB.length; i++)
			if(trie.search(setB[i]) == true)
				System.out.println("Error in insert search");
	}
	public static void testStartWith()
	{
		String words[] = { "aalii",  "aam",  "Aani", "aardvark", "aardwolf",
			"Aaronic", "Aaronite", "Aaronitic", "Aaru", "Ababdeh", "Ababua",
			"abacay", "abacinate"
		};
        String prefixes[] = {"A","a","aa","aal","Aaron","Ab","aba","abac"};
	 	String others[] = {
			"abaciscus","abacist","aback","abactinal","Abe","abaction"
		};
		Trie trie = new Trie();
		for(int i = 0; i < words.length; i++)
			trie.insert(words[i]);
		for(int i = 0; i < prefixes.length; i++)
			if(trie.startWith(prefixes[i]) != true)
				System.out.println("Error in start with");
		for(int i = 0; i < others.length; i++)
			if(trie.startWith(others[i]) != false)
				System.out.println("Error in start with");
	}
	public static void main(String[] args)
	{
		testInsertSearch();
		testStartWith();
	}
}
